### DISCUSSION TITLE
memcached throws exception at get.cc:56 Assertion "ptr->query_id == query_id + 1"

### DISCUSSION THREAD

**Discussion by Rahaneeft** [memcached/memcached#1082] [Q&A]
Hi ,
I use memcached in an multi threaded application (Running on IBM AIX).   I have one process which is responsible for writing record into memcached and there are 3 processes which are having 25 threads each which is responsible for reading. 
memcached throwing below exception while reading the record -
"    /get.cc:56 Assertion "ptr->query_id == query_id + 1" failed for function "memcached_get_by_key" likely for "Programmer error, the query_id was not incremented."     "

Version details of memcached /libmemcached given below -
Libevent library -
libevent-2.1.12-stable
https://www.monkey.org/~provos/libevent/ 

Memcached server binary -
memcached-1.6.20
https://memcached.org/downloads

Memcached client library - 
libmemcached-1.1.4

Any leads to this issue would be greatly appreciated!

Thanks!

**Comment by dormando** [comment]
Sorry, we're focused on supporting the server here and I"m not really sure what's going on there.

libmemcached isn't thread safe unless you use a specific API so be careful of accidentally using one client instance from multiple threads.
